src/net.cpp

```diff
@@ -2291,6 +2292,10 @@ void ThreadMessageHandler()
                 requester.RequestMempoolSync(syncPeer);
         }
 
+        // Periodically clear mempool sync maps in case peers have disconnected
+        if (GetStopwatchMicros() - lastMempoolSyncClear > MEMPOOLSYNC_CLEAR_FREQ_US)
+            ClearDisconnectedFromMempoolSyncMaps();
+
```

Comment:
This clearing of mempoolsync maps should really be done in FinalizeNode() in main.cpp but I don't want to hold up the PR...we could fix this in a followup.

Reply:
I don't mind moving it but can you check [this comment](https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1879#discussion_r330789893) from @gandrewstone above? FinalizeNode acts on specific node (known to be disconnecting), he seems to be suggesting that nodes can "leak" in some other way?